src/pod.c

```diff
@@ -0,0 +1,250 @@
+/*
+ * This file is part of cc-oci-runtime.
+ *
+ * Copyright (C) 2016 Intel Corporation
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/* Sandbox rootfs */
+#define CC_POD_SANDBOX_ROOTFS "workloads"
+
+/* CRI-O/ocid namespaces */
+#define CC_POD_OCID_CONTAINER_TYPE "ocid/container_type"
+#define CC_POD_OCID_SANDBOX        "infrastructure"
+#define CC_POD_OCID_CONTAINER      "container"
+
+#define CC_POD_OCID_SANDBOX_ID   "ocid/sandbox_id"
+#define CC_POD_OCID_SANDBOX_NAME "ocid/sandbox_name"
+
+#include <errno.h>
+#include <string.h>
+
+#include <glib.h>
+#include <gio/gunixconnection.h>
+
+#include "pod.h"
+#include "process.h"
+#include "proxy.h"
+#include "state.h"
+
+int
+cc_pod_handle_annotations(struct cc_oci_config *config, struct oci_cfg_annotation *annotation)
+{
+	/* We only handle CRI-O/ocid annotations for now */
+	if (strncmp(annotation->key, CC_POD_OCID_ANNOTATION_NAMESPACE,
```

Comment:
I'm not totally sure why we need `CC_POD_OCID_ANNOTATION_NAMESPACE_SIZE`? If that was dropped, you could use `g_strcmp0()`.